Assets/BossRoom/Scripts/Client/UI/Lobby/CreateLobbyUI.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Text;
+using BossRoom.Scripts.Shared.Infrastructure;
+using BossRoom.Scripts.Shared.Net.UnityServices.Lobbies;
+using Unity.Multiplayer.Samples.BossRoom;
+using Unity.Multiplayer.Samples.BossRoom.Visual;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BossRoom.Scripts.Client.UI
+{
+
+    public class CreateLobbyUI : MonoBehaviour
+    {
+        public const string k_DefaultIP = "192.168.1.1";
+        public const int k_DefaultPort = 0;
+
+        private LobbyUIManager m_LobbyUIManager;
+
+        [SerializeField]
+        private InputField m_LobbyNameInputField;
+
+        [SerializeField]
+        private InputField m_IPInputField;
+
+        [SerializeField]
+        private InputField m_PortInputField;
+
+        private OnlineMode m_OnlineMode;
+
+        #region temp
+
+        [SerializeField]
+        [Tooltip("The Animating \"Connecting\" Image you want to animate to show the client is doing something")]
+        private GameObject m_LoadingImage;
+
+        [SerializeField]
+        private Button m_ConfirmationButton;
+
+        [SerializeField]
+        private NameDisplay m_NameDisplay;
+
+        [SerializeField] private Toggle m_IPToggle;
+        [SerializeField] private Toggle m_UnityRelayToggle;
+
+        [SerializeField] private Toggle m_IsPrivate;
+
+        private static readonly char[] k_InputFieldIncludeChars = new[] {'.', '_'};
+
+
+        /// <summary>
+        /// Sanitize user port InputField box allowing only alphanumerics, plus any matching chars, if provided.
+        /// </summary>
+        /// <param name="dirtyString"> string to sanitize. </param>
+        /// <param name="includeChars"> Array of chars to include. </param>
+        /// <returns> Sanitized text string. </returns>
+        private static string Sanitize(string dirtyString, char[] includeChars = null)
+        {
+            var result = new StringBuilder(dirtyString.Length);
+            foreach (char c in dirtyString)
+            {
+                if (char.IsLetterOrDigit(c) ||
+                    (includeChars != null && Array.Exists(includeChars, includeChar => includeChar == c)))
+                {
+                    result.Append(c);
+                }
+            }
+
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Added to the InputField component's OnValueChanged callback for the Room/IP UI text.
+        /// </summary>
+        public void SanitizeInputText()
+        {
+            var inputFieldText = Sanitize(m_IPInputField.text, k_InputFieldIncludeChars);
+            m_IPInputField.text = inputFieldText;
```

Comment:
Is the temporary variable necessary?
```suggestion
            m_IPInputField.text = Sanitize(m_IPInputField.text, k_InputFieldIncludeChars);
```